src/rascal/representations/calculator_spherical_expansion.hh

```diff
@@ -2181,6 +2191,67 @@ namespace rascal {
       expansions_coefficients_gradient.set_shape(ThreeD * n_row, n_col);
     }
 
+    // species in projection matrices have to be checked here, because
+    // calculators do not have access to manager during construction
+    if (hypers.count("radial_contribution") &&
+        hypers.at("radial_contribution").get<json>().count("optimization") &&
+        hypers.at("radial_contribution")
+            .get<json>()
+            .at("optimization")
+            .get<json>()
+            .count("RadialDimReduction")) {
+      Key_t keys{};
+      for (auto center : manager) {
+        typename Key_t::value_type center_type{center.get_atom_type()};
+        keys.push_back(center_type);
+      }
+      Key_t missing_keys{};
+
+      std::map<std::string, std::vector<std::vector<std::vector<double>>>>
+          json_projection_matrices{};
+      try {
+        json_projection_matrices =
+            hypers.at("radial_contribution")
+                .get<json>()
+                .at("optimization")
+                .get<json>()
+                .at("RadialDimReduction")
+                .get<json>()
+                .at("projection_matrices")
+                .template get<
+                    std::map<std::string,
+                             std::vector<std::vector<std::vector<double>>>>>();
+      } catch (const std::exception & e) {
+        std::stringstream err_str{};
+        err_str
+            << "Projection matrices should be of type std::map<std::string, "
+               "std::vector<std::vector<std::vector<double>>>>.";
+        throw std::logic_error(err_str.str());
+      }
+      std::vector<int> projection_matrices_species{};
+      for (auto it = json_projection_matrices.begin();
+           it != json_projection_matrices.end(); ++it) {
+        projection_matrices_species.push_back(std::stoi(it->first));
+      }
```

Comment:
in `set_hyperparameters()` you already access this data. Why not set `projection_matrices_species` there if `RadialDimReduction` and use the fact that if `projection_matrices_species.count()>0` then `RadialDimReduction` was set or something like this with a simple boolea ?